joern-cli/frontends/swiftsrc2cpg/src/test/scala/io/joern/swiftsrc2cpg/passes/ast/ExtensionMultiModuleTests.scala

```diff
@@ -0,0 +1,64 @@
+package io.joern.swiftsrc2cpg.passes.ast
+
+import io.joern.swiftsrc2cpg.testfixtures.SwiftCompilerMultiModuleSrc2CpgSuite
+import io.shiftleft.semanticcpg.language.*
+
+class ExtensionMultiModuleTests extends SwiftCompilerMultiModuleSrc2CpgSuite {
+
+  private val mainCode = """
+     |import ModuleA
+     |import ModuleB
+     |
+     |func main() {
+     |    Foo().foo()
+     |    Foo().bar()
+     |}
+     |""".stripMargin
+
+  private val moduleACode =
+    """
+      |public class Foo {
+      |    public init() {}
+      |}
+      |
+      |public extension Foo {
+      |  func foo() {}
+      |}
+      |""".stripMargin
+
+  private val moduleBCode =
+    """
+      |import ModuleA
+      |
+      |public extension Foo {
+      |  func bar() {}
+      |}
+      |""".stripMargin
+
+  private val cpg =
+    codeWithSwiftSetup(mainCode, "SwiftTest/Sources/SwiftTest/main.swift")
+      .moreCode(moduleACode, "SwiftTest/Sources/ModuleA/Foo.swift")
+      .moreCode(moduleBCode, "SwiftTest/Sources/ModuleB/FooExt.swift")
+
+  "ExtensionMultiModuleTests" should {
+
+    "create correct extension method fullNames" in {
+      val List(fooTypeDecl) = cpg.typeDecl.nameExact("Foo").l
+      fooTypeDecl.fullName shouldBe "ModuleA.Foo"
+
+      val List(foo) = cpg.method.nameExact("foo").l
+      foo.fullName shouldBe "ModuleA.Foo<extension>.foo:()->()"
```

Comment:
Why does `foo` not have the full name `ModuleA.ModuleA.Foo<extension>.foo:()->()`?
Seems like you use a different naming pattern for extension in the same module as the extended type. I think a uniform naming schema is better or am I missing some reason other than the shorter full name?